crates/fluvio/src/producer/partition_producer.rs

```diff
@@ -198,11 +224,17 @@ impl PartitionProducer {
 
         let mut batch_notifiers = vec![];
 
+        let mut flush_total_bytes = 0;
+
         for p_batch in batches_ready {
             let notify = p_batch.notify.clone();
             let batch = p_batch.batch();
 
-            let raw_batch = batch.try_into()?;
+            let raw_batch: Batch<RawRecords> = batch.try_into()?;
+
+            if self.client_stats.is_some() {
+                flush_total_bytes += BATCH_FILE_HEADER_SIZE + raw_batch.records().0.len();
```

Comment:
this seems like magic.  raw_batch should return len instead of computing here

Reply:
I was calculating `batch_len` because it was getting lost in the type conversion. This is fixed in `Batch` now.